### DISCUSSION TITLE
Implement an Assign 'method'

### DISCUSSION THREAD

**Discussion by FullValueRider** [twinbasic/lang-design#24] [Ideas]
This is a Quality of Life proposal.

A common boilerplate I encounter is assigning a variant that may or may not be an object.
```
If VBA.IsObject(InVar) then
    set OutVar=InVar
Else
   OutVar = InVar
End If
```
I though I was being clever by encapsulating this boilerplate in an 'Assign' Method

```
Public Sub Assign(ByRef OutVar as Variant, ByRef InVar as Variant)
    If VBA.IsObject(InVar) then
        Set OutVar=Invar
    Else
        OutVar = InVar
    EndIf
End Sub

'So the original code above could  be written as

    Assign OutVar, Invar
```
And this mostly worked OK but I found a couple of edge cases where it did not work which made Asssign less useful than it could have been.

twinBasic has eliminated this boilerplate in the case of return values for Functions and Property Gets with  the introduction of the 'Return' keyword.
I've also seen use of the VariantCopy/VariantCopyInd  to achieve the same goal as a myAssign method (ref Olaf Scmidt's cHasdD class on VBForums)
```
Public Declare PtrSafe Function VariantCopy Lib "oleaut32" (Dst As Any, Src As Any) As Long
Public Declare PtrSafe Function VariantCopyInd Lib "oleaut32" (Dst As Any, Src As Any) As Long

' Used as

        VariantCopyInd ByVal VarPtr(mValues(i)), ByVal VarPtr(Item)
        VariantCopyInd ByVal VarPtr(mKeys(i)), ByVal VarPtr(Key)

        VariantCopy KeyByIndex, ByVal VarPtr(mKeys(IndexZeroBased))
'etc

```
Thus it would seem that there is a generic 'desire' for an easy resolution of the 'Set' boilerplate.

Consequently, I am wondering if the compiler magic that allows 'Return' in twinBasic could also be adapted to allow an Assign statement.

I fully appreciate that if used correctly, Generics and Overloaded methods could mostly eliminate the need for for such a proposal.  However in the spirit of making things easier for beginners, I feel that an 'Assign' method would be a useful and welcome addition to twinBasic.

**Comment by Greedquest** [comment]
[Related](https://codereview.stackexchange.com/q/231790/146810)

**Reply by FullValueRider** [reply]
Excellent ref.  I'd forgotten that discussion.
